.kotlin_alpha/services/firehose/src/main/kotlin/com/kotlin/firehose/StockTradeGenerator.kt

```diff
@@ -0,0 +1,124 @@
+            STOCK_PRICES.add(StockPrice("BUD", 120.39))
+            STOCK_PRICES.add(StockPrice("ORCL", 43.40))
+            STOCK_PRICES.add(StockPrice("KO", 41.23))
+            STOCK_PRICES.add(StockPrice("T", 34.64))
+            STOCK_PRICES.add(StockPrice("DIS", 101.73))
+            STOCK_PRICES.add(StockPrice("AMZN", 370.56))
+        }
+    }
+
+    private val random = Random()
+    private val id = AtomicLong(1)// add 1 because nextInt() will return between 0 (inclusive)
+
+    val randomTrade: StockTrade
+        get() {
+            // pick a random stock
+            val stockPrice = STOCK_PRICES[random.nextInt(STOCK_PRICES.size)]
+            // pick a random deviation between -MAX_DEVIATION and +MAX_DEVIATION
+            val deviation = (random.nextDouble() - 0.5) * 2.0 * MAX_DEVIATION
+            // set the price using the deviation and mean price
+            var price = stockPrice.price * (1 + deviation)
+            // round price to 2 decimal places
+            price = Math.round(price * 100.0) / 100.0
+
+            // set the trade type to buy or sell depending on the probability of sell
+            var tradeType = TradeType.BUY
+            if (random.nextDouble() < PROBABILITY_SELL) {
+                tradeType = TradeType.SELL
+            }
+
+            // randomly pick a quantity of shares
+            val quantity =
+                (random.nextInt(MAX_QUANTITY) + 1).toLong() // add 1 because nextInt() will return between 0 (inclusive)
+            // and MAX_QUANTITY (exclusive). we want at least 1 share.
+            return StockTrade(stockPrice.tickerSymbol, tradeType, price, quantity, id.getAndIncrement())
+        }
+
+    private class StockPrice internal constructor(var tickerSymbol: String, var price: Double)
+
+    /**
+     * Return a random stock trade with a unique id every time.
+     *
+     */
+
+
+    fun getSampleData(): StockTrade {
+
+        // pick a random stock
+        val stockPrice = STOCK_PRICES[random.nextInt(STOCK_PRICES.size)]
+        // pick a random deviation between -MAX_DEVIATION and +MAX_DEVIATION
+        val deviation = (random.nextDouble() - 0.5) * 2.0 * MAX_DEVIATION
+        // set the price using the deviation and mean price
+        var price = stockPrice.price * (1 + deviation)
+        // round price to 2 decimal places
+        price = Math.round(price * 100.0) / 100.0
+
+        // set the trade type to buy or sell depending on the probability of sell
+        var tradeType = TradeType.BUY
+        if (random.nextDouble() < PROBABILITY_SELL) {
+            tradeType = TradeType.SELL
+        }
+
+        // randomly pick a quantity of shares
+        val quantity =
+            random.nextInt(MAX_QUANTITY) + 1.toLong() // add 1 because nextInt() will return between 0 (inclusive)
+        // and MAX_QUANTITY (exclusive). we want at least 1 share.
+        return StockTrade(stockPrice.tickerSymbol, tradeType, price, quantity, id.getAndIncrement())
+    }

```

Comment:
Looks like the same code as randomTrade.get() above. Did you mean to replace one with the other?

Reply:
fixed